public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,814 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { Component, Fragment, useState, useEffect } from 'react';
+import { Formik } from 'formik';
+import { hasError, isInvalid, required, validatePositiveInteger } from '../../../utils/validate';
+
+import {
+  EuiFlexItem,
+  EuiText,
+  EuiFormRow,
+  EuiButton,
+  EuiPageBody,
+  EuiSpacer,
+  EuiBasicTable,
+  EuiTitle,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiButtonEmpty,
+  EuiFlyoutBody,
+  EuiCallOut,
+  EuiIcon,
+  EuiTextColor,
+  EuiPanel,
+  EuiFlexGrid,
+  EuiFormLabel,
+  EuiLink,
+  EuiPopover,
+  EuiExpression,
+} from '@elastic/eui';
+import ContentPanel from '../../ContentPanel';
+import {
+  FormikFieldText,
+  FormikFieldNumber,
+  FormikComboBox,
+  FormikSelect,
+} from '../../FormControls';
+import { KIBANA_AD_PLUGIN, DATA_TYPES, DETECTOR_CREATION_CALLOUTS } from '../../../utils/constants';
+import { getIndexFields } from '../../../pages/CreateMonitor/components/MonitorExpressions/expressions/utils/dataTypes';
+import {
+  getOperators,
+  isNullOperator,
+  isRangeOperator,
+  validateRange,
+  displayText,
+} from '../../../pages/CreateMonitor/components/MonitorExpressions/expressions/utils/whereHelpers';
+import {
+  POPOVER_STYLE,
+  EXPRESSION_STYLE,
+  WHERE_BOOLEAN_FILTERS,
+} from '../../../pages/CreateMonitor/components/MonitorExpressions/expressions/utils/constants';
+import { getPathsPerDataType } from '../../../pages/CreateMonitor/containers/DefineMonitor/utils/mappings';
+import { formikToWhereClause } from '../../../pages/CreateMonitor/containers/CreateMonitor/utils/formikToMonitor';
+import { MAX_INTERVAl_LENGTH_MINUTES } from '../../../pages/MonitorDetails/containers/utils/helpers';
+
+function toString(obj) {
```

Comment:
This function does multiple work: 1. return `period` filed of input `obj` 2.parse input `obj` to specific time format if it's number type. 3. return '-' for undefined input

Suggest to split this function into two and make the function name reveal what exactly it does.  Suggest use `lodash` get function to simplify code.